src/higley_lab_to_nwb/lohani_2022/interfaces/lohani_2022_visual_stimulus_interface.py

```diff
@@ -0,0 +1,121 @@
+    Design) using the :py:class:`~spikeinterface.extractors.CedRecordingExtractor`."""
+
+    display_name = "Spike2 Recording"
+    associated_suffixes = (".smr", ".smrx")
+    info = "Interface for Spike2 analogue signals from CED (Cambridge Electronic Design)."
+
+    def __init__(
+        self,
+        spike2_file_path: FilePathType,
+        csv_file_path: FilePathType,
+        stream_id: str,
+        verbose: bool = True,
+    ):
+        """
+         Parameters
+        ----------
+        spike2_file_path : FilePathType
+            Path to .smr or .smrx file.
+        csv_file_path : FilePathType
+            Path to .csv file for visual stimulus characterization.
+        verbose : bool, default: True
+        """
+        _test_sonpy_installation()
+
+        super().__init__(
+            spike2_file_path=spike2_file_path,
+            csv_file_path=csv_file_path,
+            stream_id=stream_id,
+            verbose=verbose,
+        )
+
+    def get_event_times_from_ttl(self, rising: bool = True):
+        extractor = CedRecordingExtractor(
+            file_path=str(self.source_data["spike2_file_path"]), stream_id=self.source_data["stream_id"]
+        )
+        times = extractor.get_times()
+        traces = extractor.get_traces()
+        if rising:
+            event_times = get_rising_frames_from_ttl(traces)
+        else:
+            event_times = get_falling_frames_from_ttl(traces)
+
+        return times[event_times]
+
+    def get_stimulus_feature(self, column_index):
+        feature = pd.read_csv(self.source_data["csv_file_path"], usecols=column_index)
+        return feature.to_numpy()
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False) -> None:
+
+        intervals_table = TimeIntervals(
+            name="VisualStimulus",
+            description=f"Intervals for each visual stimulus presentation",
+        )
+
+        intervals_table.add_column(name="contrast", description="Contrast of the visual stimulus image.")
+        contrasts = self.get_stimulus_feature(column_index=[0])
+        intervals_table.add_column(name="orientation", description="Contrast of the visual stimulus image.")
+        orientations = self.get_stimulus_feature(column_index=[1])
+        intervals_table.add_column(name="stimulus_frequency", description="Temporal frequency of the stimulus, in Hz.")
+        stimulus_frequencies = self.get_stimulus_feature(column_index=[2])
+        intervals_table.add_column(
+            name="spatial_frequency", description="Spatial frequency of the stimulus, in cycles per degrees."
+        )
+        spatial_frequencies = self.get_stimulus_feature(column_index=[3])
+        intervals_table.add_column(name="size", description="Size of the visual stimulus, in degrees.")
+        sizes = self.get_stimulus_feature(column_index=[4])
+        intervals_table.add_column(name="screen_coordinates", description="Visual stimulus coordinates on the screen.")
```

Comment:
Do we know what unit are they in? I think a little bit more information would be useful here

Reply:
Also the order of coordinates would be useful to add

Reply:
NO, exactly, I need to ask them. I forgot to write a TODO comment